src/components/Latex/Latex.tsx

```diff
@@ -0,0 +1,48 @@
+import './Latex.css'
+
+import {MathComponent} from 'mathjax-react'
+import React from 'react'
+
+let re = /((?<!(\\|\$))\$(?!\$).+?(?<!(\\|\$))\$(?!\$))|(\$\$.+?\$\$)|(\\\[.+?\\\])|(\\\(.+?\\\))/gs
+
+let trim = (str: string) => {
+  if (str[0] === '$' && str[1] !== '$') {
+    return str.substring(1, str.length - 1)
+  } else {
+    return str.substring(2, str.length - 2)
+  }
+}
+
+export const Latex: React.FC<{children: any}> = ({children}) => {
+  let matches: any = Array.from(children.matchAll(re))
+
+  if (matches.length === 0) {
+    return <>{children}</>
+  }
+
+  let result = []
+  let currentPosition: number = 0
+
+  for (let m in matches) {
+    result.push(<>{children.substring(currentPosition, matches[m].index)}</>)
+    if (matches[m][0].substring(0, 2) === '\\[' || matches[m][0].substring(0, 2) === '$$') {
+      result.push(
+        <>
+          <MathComponent tex={trim(matches[m][0])} display={true} />
+        </>,
+      )
+    } else {
+      result.push(
+        <>
+          <MathComponent tex={trim(matches[m][0])} display={false} />
+        </>,
+      )
+    }
```

Comment:
kedze obe branche vyzeraju skoro rovnako, dalo by sa to skratit 😄 
```suggestion
  result.push(
      <>
        <MathComponent
          tex={trim(matches[m][0])}
          display={matches[m][0].substring(0, 2) === '\\[' || matches[m][0].substring(0, 2) === '$$'}
        />
      </>,
    )
```
a/alebo kludne vynat do osobitnej premennej `const display = matches[m][0].substring(0, 2) === '\\[' || matches[m][0].substring(0, 2) === '$$'` alebo dokonca aj `const match = matches[m][0]`.
jo a ten fragment je tiez zbytocny 😄 takze dostaneme:
```js
    const match = matches[m][0]
    const display = match.substring(0, 2) === '\\[' || match.substring(0, 2) === '$$'
    result.push(<MathComponent tex={trim(match)} display={display} />)
```